platforms/core-configuration/core-serialization-codecs/src/main/kotlin/org/gradle/internal/serialize/codecs/core/WorkNodeCodec.kt

```diff
@@ -300,6 +300,16 @@ class WorkNodeCodec(
                 writeSmallInt(nodeId)
                 safeRun {
                     write(node)
+                    if (node is ActionNode) {
+                        // could potentially trigger expensive computation, so even though this is not
+                        // related to node state, trigger it here to piggyback on parallel CC node store;
+                        // results are consumed later via postExecutionNodes
+                        val setupNodeAction = node.action?.preExecutionNode
+                        // Could probably add some abstraction for nodes that can be executed eagerly and discarded
```

Comment:
💭 Perhaps an inline method that runs the inner action guarded by the typecheck used in both places would be a good way to keep the connection for now? e.g. `withPostExecutionNodeAware(setupNodeAction) { runPre...(it) }`